organization.go

```diff
@@ -48,12 +48,20 @@ type Organizations interface {
 
 	// ReadDataRetentionPolicy reads an organization's data retention policy
 	// **Note: This functionality is only available in Terraform Enterprise.**
-	ReadDataRetentionPolicy(ctx context.Context, organization string) (*DataRetentionPolicy, error)
+	ReadDataRetentionPolicy(ctx context.Context, organization string) (*DataRetentionPolicyChoice, error)
 
-	// SetDataRetentionPolicy sets an organization's data retention policy
-	// **Note: This functionality is only available in Terraform Enterprise.**
+	// DEPRECATED: Use SetDataRetentionPolicyDeleteOlder instead
+	// **Note: This functionality is only available in Terraform Enterprise versions v202311-1 and v202312-1.**
 	SetDataRetentionPolicy(ctx context.Context, organization string, options DataRetentionPolicySetOptions) (*DataRetentionPolicy, error)
 
+	// SetDataRetentionPolicyDeleteOlder sets an organization's data retention policy to delete data older than a certain number of days
+	// **Note: This functionality is only available in Terraform Enterprise.**
+	SetDataRetentionPolicyDeleteOlder(ctx context.Context, organization string, options DataRetentionPolicyDeleteOlderSetOptions) (*DataRetentionPolicyDeleteOlder, error)
+
+	// SetDataRetentionPolicyDontDelete sets an organization's data retention policy to explicitly not delete data
+	// **Note: This functionality is only available in Terraform Enterprise.**
+	SetDataRetentionPolicyDontDelete(ctx context.Context, organization string, options DataRetentionPolicyDontDeleteSetOptions) (*DataRetentionPolicyDontDelete, error)
```

Comment:
`DataRetentionPolicyDontDeleteSetOptions` is always empty here, so the argument could easily be omitted...I included it so that if options were added in the future the same function signature would still be valid.

Very open to just removing the argument if that is preferred though